vault/seal/seal_testing.go

```diff
@@ -34,26 +35,37 @@ func NewToggleableTestSeal(opts *TestSealOpts) (*Access, *error) {
 	return &Access{
 		Wrapper:        w,
 		OverriddenType: opts.Name,
-	}, &w.Error
+	}, w.SetError
 }
 
 type ToggleableWrapper struct {
 	wrapping.Wrapper
-	Error error
+	error error
+	l     sync.RWMutex
 }
 
-func (t ToggleableWrapper) Encrypt(ctx context.Context, bytes []byte, bytes2 []byte) (*wrapping.EncryptedBlobInfo, error) {
-	if t.Error != nil {
-		return nil, t.Error
+func (t *ToggleableWrapper) Encrypt(ctx context.Context, bytes []byte, bytes2 []byte) (*wrapping.EncryptedBlobInfo, error) {
+	t.l.RLock()
+	t.l.RUnlock()
```

Comment:
Missing the defer keyword here and on line 58?

Reply:
thx